backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/process/adprocess/ADProcessPostProcessService.java

```diff
@@ -126,10 +130,19 @@ private void invalidateDocumentsAndViews(final ViewId viewId, final ProcessExecu
 		//
 		// Refresh all
 		boolean viewInvalidateAllCalled = false;
-		if (processExecutionResult.isRefreshAllAfterExecution() && viewSupplier.get() != null)
+		if (processExecutionResult.isRefreshAllAfterExecution())
 		{
-			viewSupplier.get().invalidateAll();
-			viewInvalidateAllCalled = true;
+			if (viewSupplier.get() != null)
+			{
+				viewSupplier.get().invalidateAll();
+				viewInvalidateAllCalled = true;
+			}
+			else if (currentSingleSelectedDocumentRef != null)
+			{
+				documentsCollection.invalidateDocumentByRecordId(
```

Comment:
wanted to call, but i think it's better to list my questions..
* since we are in the `isRefreshAllAfterExecution()` branch, don't we have to find a way to invalidate all records?
* in case of `isRefreshAllAfterExecution() == false`, shouldn't we *still* make sure to invalidate the selected record?
* in the dev-test you wrote "select a product and run the process". So, i can imagine the test succeeds with a *single* record. but what if more than one record is selected?
* maybe consult with Teo?

Reply:
One short, the case of more than one record selected is already covered by previous lines:
![image](https://user-images.githubusercontent.com/15378036/83519221-424c0b00-a4e4-11ea-93be-bb82e03902f4.png)

>in case of isRefreshAllAfterExecution() == false, shouldn't we still make sure to invalidate the selected record?

Hm.. it is wanted? Not sure.